wandb/sdk/internal/job_builder.py

```diff
@@ -76,6 +78,7 @@ class JobBuilder:
     _summary: Optional[Dict[str, Any]]
     _logged_code_artifact: Optional[ArtifactInfoForJob]
     _disable: bool
+    _partial: Optional["UseArtifactRecord"]
```

Comment:
Wouldn't it make more sense to have _partial be a `Union[GitSourceDict, ArtifactSourceDict, ImageSourceDict]`

Then you can just convert the proto to the right source dict, and then skip making the source dict, and instead just create an artifact and inject teh files?

Reply:
Absolutely, but we need additional information like runtime/python right? I can get it closer, but we will always need more information when making the partial job the first time. @KyleGoyette

Reply:
Maybe I'm misunderstanding, I'll try exactly that.

Reply:
Yeah, we can do even better, we could have `_partial` be a SourceDict with the input and output types omitted (the rest of the info is already available in the protobuf)